src/durable-objects/CampaignManager.ts

```diff
@@ -0,0 +1,161 @@
+    }
+  }
+
+  private async createCampaign(request: Request): Promise<Response> {
+    const body = await request.json();
+    if (
+      !body ||
+      typeof body !== "object" ||
+      !("name" in body) ||
+      typeof (body as Record<string, unknown>).name !== "string"
+    ) {
+      return new Response(
+        JSON.stringify({ error: "Invalid request: name is required." }),
+        { status: 400, headers: { "Content-Type": "application/json" } }
+      );
+    }
+    const { name } = body as { name: string };
+    const campaignId = crypto.randomUUID();
+    const now = new Date().toISOString();
+    this.campaign = {
+      campaignId,
+      name,
+      createdAt: now,
+      updatedAt: now,
+      resources: [],
+    };
+    console.debug(`[CampaignManager] Created campaign ${campaignId}`);
+    return new Response(
+      JSON.stringify({ success: true, campaign: this.campaign }),
+      {
+        headers: { "Content-Type": "application/json" },
+      }
+    );
+  }
+
+  private async addResource(request: Request): Promise<Response> {
+    if (!this.campaign) return new Response("No campaign", { status: 404 });
+    const resource = await request.json();
+    if (
+      !resource ||
+      typeof resource !== "object" ||
+      !("type" in resource) ||
+      !("id" in resource) ||
+      typeof (resource as Record<string, unknown>).type !== "string" ||
+      typeof (resource as Record<string, unknown>).id !== "string"
+    ) {
+      return new Response(
+        JSON.stringify({
+          error: "Invalid resource: type and id are required.",
+        }),
+        { status: 400, headers: { "Content-Type": "application/json" } }
+      );
+    }
+    this.campaign.resources.push(resource as CampaignResource);
+    this.campaign.updatedAt = new Date().toISOString();
+    console.debug(
+      `[CampaignManager] Added resource to ${this.campaign.campaignId}:`,
+      resource
+    );
+    return new Response(
+      JSON.stringify({ success: true, resources: this.campaign.resources }),
+      {
+        headers: { "Content-Type": "application/json" },
+      }
+    );
+  }
+
+  private async removeResource(request: Request): Promise<Response> {
+    if (!this.campaign) return new Response("No campaign", { status: 404 });
+    const url = new URL(request.url);
+    const parts = url.pathname.split("/");
+    const resourceId = parts[parts.length - 1];
+
+    // Check if the resource exists before attempting to remove it
+    const resourceExists = this.campaign.resources.some(
+      (r) => r.id === resourceId
+    );
+    if (!resourceExists) {
+      return new Response(
+        JSON.stringify({
+          error: `Resource with ID '${resourceId}' not found in campaign.`,
+        }),
+        {
+          status: 404,
+          headers: { "Content-Type": "application/json" },
+        }
+      );
+    }
+
+    this.campaign.resources = this.campaign.resources.filter(
```

Comment:
should have type checking -- something like `if (!this.campaign.resources) this.campaign.resources = {}` (i assume `campaign is safe, but also check that if not